src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseGradingService.java

```diff
@@ -652,20 +636,19 @@ private void addFeedbackTestsNotExecuted(final Result result, final ProgrammingE
         removeAllTestCaseFeedbackAndSetScoreToZero(result, staticCodeAnalysisFeedback);
 
         // Add feedbacks for all duplicate test cases
-        createFeedbackForDuplicateTests(result, exercise);
+        createFeedbacksForDuplicateTests(result, exercise);
     }
 
     /**
      * Only keeps automatic feedbacks that also are associated with a test case.
      * <p>
      * Does not remove static code analysis feedback.
      *
-     * @param result of the build run.
+     * @param result    of the build run.
      * @param testCases of the programming exercise.
      */
-    private void retainAutomaticFeedbacksWithTestCase(Result result, final Set<ProgrammingExerciseTestCase> testCases) {
-        // Remove automatic feedbacks not associated with test cases
-        result.getFeedbacks().removeIf(feedback -> feedback.getType() == FeedbackType.AUTOMATIC && !feedback.isStaticCodeAnalysisFeedback()
+    private void filterAutomaticFeedbacksWithoutTestCase(Result result, final Set<ProgrammingExerciseTestCase> testCases) {
+        result.getFeedbacks().removeIf(feedback -> feedback.getType() == FeedbackType.AUTOMATIC && !feedback.isStaticCodeAnalysisFeedback()//
```

Comment:
I guess this is to stop the automatic formatter (is it?). In this case we should use ` @formatter:off` and ` @formatter:on`. same below.
```suggestion
        result.getFeedbacks().removeIf(feedback -> feedback.getType() == FeedbackType.AUTOMATIC && !feedback.isStaticCodeAnalysisFeedback()
```

Reply:
My suggested alternative would be to just accept how the formatter wants to have the code.
The formatter:off is usually only used in Artemis for long builder pattern chains like in the Spring security configuration. Otherwise I think we should avoid the manual formatting.

Reply:
Well, time to buy an ultrawide monitor